rand_distr/src/normal.rs

```diff
@@ -140,23 +144,57 @@ impl std::error::Error for Error {}
 impl<F> Normal<F>
 where F: Float, StandardNormal: Distribution<F>
 {
-    /// Construct a new `Normal` distribution with the given mean and
-    /// standard deviation.
+    /// Construct, from mean and standard deviation
+    ///
+    /// Parameters:
+    ///
+    /// -   mean (`μ`, unrestricted)
+    /// -   standard deviation (`σ`, must be finite)
     #[inline]
     pub fn new(mean: F, std_dev: F) -> Result<Normal<F>, Error> {
-        if !(std_dev >= F::zero()) {
-            return Err(Error::StdDevTooSmall);
+        if !std_dev.is_finite() {
+            return Err(Error::BadVariance);
         }
         Ok(Normal { mean, std_dev })
     }
+
+    /// Construct, from mean and coefficient of variation
+    ///
+    /// Parameters:
+    ///
+    /// -   mean (`μ`, unrestricted)
+    /// -   coefficient of variation (`cv = abs(σ / μ)`)
+    #[inline]
+    pub fn from_mean_cv(mean: F, cv: F) -> Result<Normal<F>, Error> {
+        if !cv.is_finite() || cv < F::zero() {
+            return Err(Error::BadVariance);
+        }
+        let std_dev = cv * mean;
+        Ok(Normal { mean, std_dev })
+    }
+
+    /// Sample from a z-score
+    ///
+    /// This may be useful for generating correlated samples, as follows.
+    /// ```
+    /// # use rand::prelude::*;
+    /// # use rand_distr::{Normal, StandardNormal};
+    /// let mut rng = thread_rng();
+    /// let z = StandardNormal.sample(&mut rng);
+    /// let x1 = Normal::new(0.0, 1.0).unwrap().from_zscore(z);
+    /// let x2 = Normal::new(2.0, -1.0).unwrap().from_zscore(z);
+    /// ```
```

Comment:
```suggestion
    /// This may be useful for generating correlated samples `x1` and `x2`
    /// from two different distributions, as follows.
    /// ```
    /// # use rand::prelude::*;
    /// # use rand_distr::{Normal, StandardNormal};
    /// let mut rng = thread_rng();
    /// let z = StandardNormal.sample(&mut rng);
    /// let x1 = Normal::new(0.0, 1.0).unwrap().from_zscore(z);
    /// let x2 = Normal::new(2.0, -3.0).unwrap().from_zscore(z);
    /// ```
```